extensions/amp-analytics/0.1/instrumentation.js

```diff
@@ -212,14 +213,21 @@ export class InstrumentationService {
    * @private
    */
   createVisibilityListener_(callback, config) {
-    if (config['visibilitySpec']) {
+    const spec = config['visibilitySpec'];
+    if (spec) {
       if (!isVisibilitySpecValid(config)) {
         return;
       }
+      const selector = config['selector'] || (spec && spec['selector']);
+      const analyticsVariables = config[VARS] || {};
+      const el = this.win_.document.getElementById(selector
+        .slice(1));
+      Object.assign(analyticsVariables, this.extractAnalyticsVariables_(el));
+      config[VARS] = analyticsVariables;
 
       this.runOrSchedule_(() => {
         visibilityFor(this.win_).then(visibility => {
-          visibility.listenOnce(config['visibilitySpec'], vars => {
+          visibility.listenOnce(config, vars => {
             callback(new AnalyticsEvent(AnalyticsEventType.VISIBLE, vars));
```

Comment:
For `visibility.listenOnce` we need both the `visibilitySpec` and the merged `vars`. Merging of vars is done on #L225